### PR TITLE
Random crop migrate

### PR DESCRIPTION
**Pull request by divyashreepathihalli**

### CODE REVIEW

**Comment by divyashreepathihalli** [comment]
random_crop_demo_dec_2022

File: keras_cv/layers/preprocessing/random_crop.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+    def __init__(self, height, width, seed=None, bounding_box_format=None, **kwargs):
+        super().__init__(**kwargs, autocast=False, seed=seed, force_generator=True)
+        self.height = height
+        self.width = width
+        self.seed = seed
+        self.auto_vectorize = False
+        self.bounding_box_format = bounding_box_format
+
+    def call(self, inputs, training=True):
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/random_crop.py:64]: Do we need to override `call`? 

It seems like we should be able to omit this and rely on the BaseImageAugmentationLayer

File: keras_cv/layers/preprocessing/random_crop.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        # Divya to-do add support in follow up PR
+        return bounding_boxes
+
+    def compute_output_shape(self, input_shape):
+        input_shape = tf.TensorShape(input_shape).as_list()
+        input_shape[H_AXIS] = self.height
+        input_shape[W_AXIS] = self.width
+        return tf.TensorShape(input_shape)
+
+    def get_config(self):
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/random_crop.py:125]: Let's add this layer to layers/serialization_test.py

File: keras_cv/layers/preprocessing/random_crop_test.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            actual_output = layer(inp, training=True)
+            expected_output = inp[
+                :,
+                height_offset : (height_offset + height),
+                width_offset : (width_offset + width),
+                :,
+            ]
+            self.assertAllClose(expected_output, actual_output)
+
+    @parameterized.named_parameters(
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/random_crop_test.py:79]: It seems like all of these tests which delegate to `self._run_test` could just be expressed as parameterizations of `_run_test`, and that test could be called `test_output_shape`

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thanks!

File: keras_cv/layers/preprocessing/random_crop.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        resizing_layer = tf.keras.layers.Resizing(self.height, self.width)
+        outputs = resizing_layer(image)
+        # smart_resize will always output float32, so we need to re-cast.
+        return tf.cast(outputs, self.compute_dtype)
+
+    def augment_label(self, label, transformation, **kwargs):
+        return label
+
+    def augment_bounding_boxes(self, bounding_boxes, transformation=None, **kwargs):
+        # Divya to-do add support in follow up PR
```
[the remark this answers is not in the document]
Comment (divyashreepathihalli) [reply keras_cv/layers/preprocessing/random_crop.py]: Removing this. I will add the bbox support in followup PR

File: keras_cv/layers/preprocessing/random_crop.py (hunk 2 above)
Comment (divyashreepathihalli) [reply keras_cv/layers/preprocessing/random_crop.py:104]: Done!

File: keras_cv/layers/preprocessing/random_crop.py (hunk 1 above)
Comment (divyashreepathihalli) [reply keras_cv/layers/preprocessing/random_crop.py]: Removed!

File: keras_cv/layers/preprocessing/random_crop_test.py (hunk 3 above)
Comment (divyashreepathihalli) [reply keras_cv/layers/preprocessing/random_crop_test.py:79]: done!

File: keras_cv/layers/preprocessing/random_crop.py [hunk 5]
```diff
@@ -112,10 +95,6 @@ def _resize(self, image):
     def augment_label(self, label, transformation, **kwargs):
         return label
 
-    def augment_bounding_boxes(self, bounding_boxes, transformation=None, **kwargs):
-        # Divya to-do add support in follow up PR
-        return bounding_boxes
-
     def compute_output_shape(self, input_shape):
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/random_crop.py:98]: One other bit -- I think we should override `compute_image_signature` instead -- I don't think this method does anything in our Base layer

**Review by ianstenbit — APPROVED** [review]
LGTM mod one more comment -- thanks!

**Comment by divyashreepathihalli** [comment]
Unable to reopen the PR, Sorry about the messup - a new PR here - https://github.com/keras-team/keras-cv/pull/1141

**Pull request closed without merging**